crates/hyperdrive-math/src/long/open.rs

```diff
@@ -36,21 +37,29 @@ impl State {
         };
         let max_spot_price = self.get_max_spot_price();
         if ending_spot_price > max_spot_price {
-            // TODO would be nice to return a `Result` here instead of a panic.
-            panic!("InsufficientLiquidity: Negative Interest");
+            return Err(eyre!(
+                "calculate_open_long: InsufficientLiquidity: Negative Interest",
+            ));
         }
 
-        long_amount - self.open_long_curve_fees(base_amount)
+        Ok(long_amount - self.open_long_curve_fees(base_amount))
     }
 
     #[deprecated(since = "0.4.0", note = "please use `calculate_open_long` instead")]
     pub fn get_long_amount<F: Into<FixedPoint>>(&self, base_amount: F) -> FixedPoint {
-        self.calculate_open_long(base_amount)
+        self.calculate_open_long(base_amount).unwrap()
```

Comment:
Any reason not to have this return a `Result` also? We should at least leave a `TODO`.

Reply:
If you'd like I can change all the Options to Results, that makes more sense to me I was just trying to follow patterns I found elsewhere.